packages/flutter/lib/src/material/switch.dart

```diff
@@ -281,6 +283,41 @@ class Switch extends StatelessWidget {
   /// | Disabled | `Colors.grey.shade400`            | `Colors.grey.shade800`            |
   final MaterialStateProperty<Color?>? thumbColor;
 
+  /// {@template flutter.material.switch.thumbImage}
+  /// The image of this [Switch]'s thumb.
+  ///
+  /// Resolved in the following states:
+  ///  * [MaterialState.selected].
+  ///  * [MaterialState.hovered].
+  ///  * [MaterialState.focused].
+  ///  * [MaterialState.disabled].
+  ///
+  /// {@tool snippet}
+  /// This example resolves the [thumbImage] based on the current
+  /// [MaterialState] of the [Switch], providing a different [ImageProvider] when it is
+  /// [MaterialState.disabled].
+  ///
+  /// ```dart
+  /// Switch(
+  ///   value: true,
+  ///   onChanged: (_) => true,
+  ///   thumbColor: MaterialStateProperty.resolveWith<ImageProvider>((Set<MaterialState> states) {
```

Comment:
I think you want a thumbImage example here

Reply:
Done ✔️